pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found.' % namespace)
+
+        qname = params['QualifierName']
+
+        if qname in self.qualifiers[namespace]:
+            del self.qualifiers[namespace][qname]
+        else:
+            raise CIMError(CIM_ERR_NOT_FOUND, 'Qualifier decl %s not found.'
+                           % qname)
+
+    #####################################################################
+    #
+    #  Faked WBEMConnection Instance methods
+    #
+    #####################################################################
+
+    def _fake_createinstance(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`~pywbem.WBEMConnection.CreateInstance`
+
+        Create a CIM instance in the local repository of this class.
+
+        Exceptions:
+            CIMError CIM_ERR_ALREADY_EXISTS, CIM_ERR_INVALID_CLASS
+        """
+
+        new_inst = params['NewInstance']
+
+        # Requires corresponding class to build path to be returned
+        if not self._class_exists(namespace, new_inst.classname):
+            raise CIMError(CIM_ERR_INVALID_CLASS, 'Cannot create instance '
+                                                  'without corresponding class '
+                                                  '%s.' % new_inst.classname)
+        # TODO change this to use std getclass
+        class_for_inst = self.classes[namespace][new_inst.classname]
+
+        # If property not in instance, add it and use default value from
+        # class
+        for cprop_name in class_for_inst.properties:
+            if cprop_name not in new_inst:
+                default_value = class_for_inst.properties[cprop_name]
+                new_inst[cprop_name] = default_value
+
+        # Exception if property in instance but not class
+        for iprop in new_inst:
+            if iprop not in class_for_inst.properties:
+                raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                               'Property %s in instance but not in class %s'
+                               % (iprop, class_for_inst))
+
+        # if namespace in instance.path use it.
+        path = self.create_instance_path(class_for_inst, new_inst, namespace)
+        new_inst.path = path
+        try:
+            for inst in self.instances[namespace]:
+                if inst.path == new_inst.path:
+                    raise CIMError(CIM_ERR_ALREADY_EXISTS,
+                                   'Instance %s exists in namespace %s' %
+                                   (new_inst.path, namespace))
+            self.instances[namespace].append(new_inst)
+        except KeyError:
+            self.instances[namespace] = [new_inst]
+        # Create instance returns model path, path relative to namespace
+        # TODO per spec. path returned if any keys are dynamically allocated.
+        return self._make_tuple([copy.deepcopy(path)])
```

Comment:
`CIMInstanceName` has a `copy()` method. I suggest using that instead of `copy.deepcopy()`.

Reply:
Done